torchaudio/prototype/tacotron2.py

```diff
@@ -0,0 +1,847 @@
+from typing import Tuple, List, Optional, Union
+
+import torch
+from torch import nn
+from torch import Tensor
+from torch.nn import functional as F
+
+
+__all__ = [
+    "Tacotron2",
+]
+
+
+class _LinearNorm(torch.nn.Module):
+    r"""Linear layer with xavier uniform initialization
+
+    Args:
+        in_dim (int): Size of each input sample.
+        out_dim (int): Size of each output sample.
+        bias (bool, optional): If set to ``False``, the layer will not learn an additive bias. (Default: ``True``)
+        w_init_gain (str, optional): Parameter passed to ``torch.nn.init.calculate_gain``
+            for setting the gain parameter of ``xavier_uniform_``. (Default: ``linear``)
+    """
+    def __init__(self, in_dim: int, out_dim: int, bias: bool = True,
+                 w_init_gain: str = 'linear') -> None:
+        super().__init__()
+        self.linear_layer = torch.nn.Linear(in_dim, out_dim, bias=bias)
+
+        torch.nn.init.xavier_uniform_(
+            self.linear_layer.weight,
+            gain=torch.nn.init.calculate_gain(w_init_gain))
+
+    def forward(self, x: Tensor) -> Tensor:
+        return self.linear_layer(x)
+
+
+class _ConvNorm(torch.nn.Module):
+    r"""1D convolution with xavier uniform initialization
+
+    Args:
+        in_channels (int): Number of channels in the input image.
+        out_channels (int): Number of channels produced by the convolution.
+        kernel_size (int, optional): Number of channels in the input image. (Default: ``1``)
+        stride (int, optional): Number of channels in the input image. (Default: ``1``)
+        padding (str, int or tuple, optional): Padding added to both sides of the input.
+            (Default: dilation * (kernel_size - 1) / 2)
+        dilation (int, optional): Number of channels in the input image. (Default: ``1``)
+        w_init_gain (str, optional): Parameter passed to ``torch.nn.init.calculate_gain``
+            for setting the gain parameter of ``xavier_uniform_``. (Default: ``linear``)
+    """
+    def __init__(self, in_channels: int, out_channels: int, kernel_size: int = 1, stride: int = 1,
+                 padding: Optional[Union[str, int, Tuple[int]]] = None, dilation: int = 1,
+                 bias: bool = True, w_init_gain: str = 'linear') -> None:
+        super().__init__()
+        if padding is None:
+            assert (kernel_size % 2 == 1)
+            padding = int(dilation * (kernel_size - 1) / 2)
+
+        self.conv = torch.nn.Conv1d(in_channels, out_channels,
+                                    kernel_size=kernel_size, stride=stride,
+                                    padding=padding, dilation=dilation,
+                                    bias=bias)
+
+        torch.nn.init.xavier_uniform_(
+            self.conv.weight,
+            gain=torch.nn.init.calculate_gain(w_init_gain))
+
+    def forward(self, signal):
+        return self.conv(signal)
+
+
+def get_mask_from_lengths(lengths: Tensor) -> Tensor:
```

Comment:
nit: since other non-public interfaces are prefixed with underscore, this function looks public API. (I am guessing it is not)

Reply:
Thanks for pointing it out. It's addressed [here](https://github.com/pytorch/audio/pull/1621/commits/68b127377f8f8f8b6ae64c06ff4f6de885335d57#diff-8b8ec57192b9aaeb99a9ec5e709c977771b31b49fe7ddc6704c2128cf161a00eR97)